### PR TITLE
Feature/p2022 972 refactor localized link component

### PR DESCRIPTION
**Pull request by Reden4005**
Refactor localized link component.

### CODE REVIEW

File: libs/shared/src/types/app-routes.ts [hunk 1]
```diff
@@ -7,3 +7,21 @@ export enum AppRoute {
...
+  Home = "home",
+  Presentation = "presentation",
+  AddPresentation = "add-presentation",
+  EditPresentation = "edit-presentation",
+  PresentationForExternalUser = "external-user-presentation",
+  Dashboard = "dashboard"
+}
+
+export const PAGE_PATHS: Record<PagePath, AppRoute | AppRoute[]> = {
+  [PagePath.Dashboard]: AppRoute.Dashboard,
```
Comment (pawelwojdas) [inline libs/shared/src/types/app-routes.ts:21]: PAGE_PATHS are constants, not types. Shouldn't we move it to other directory?

File: libs/shared/src/components/localized-link/index.tsx [hunk 2]
```diff
@@ -1,16 +1,19 @@
-import { AppRoute } from "@patronage-web/shared";
 import { useTranslation } from "react-i18next";
-import { generatePath, Link, Params } from "react-router-dom";
+import { Link, Params } from "react-router-dom";
+
+import { PAGE_PATHS, PagePath } from "../../types";
+import { createPath } from "../../utils";
```
Comment (pawelwojdas) [inline libs/shared/src/components/localized-link/index.tsx:5]: Shouldn't we import as: `import { createPath, PAGE_PATHS, PagePath } from "@patronage-web/shared";`

File: libs/shared/src/utils/create-path.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { generatePath, Params } from "react-router-dom";
+
+import { AppRoute, SupportedLanguage } from "../types";
+
+export const createPath = (path: AppRoute | AppRoute[], params?: Params, language?: string) => {
+  const lang = language?.length && language !== SupportedLanguage.En ? `?lang=${language}` : "";
+  const basePath = path instanceof Array ? path.join("/") : path;
```
Comment (kacperChlon) [inline libs/shared/src/utils/create-path.ts:7]: Please add an empty space before return

File: libs/shared/src/types/app-routes.ts (hunk 1 above)
Comment (kacperChlon) [reply libs/shared/src/types/app-routes.ts:21]: Yes, we can ;) to constants/index.ts

File: libs/shared/src/components/localized-link/index.tsx (hunk 2 above)
Comment (kacperChlon) [reply libs/shared/src/components/localized-link/index.tsx:5]: No, and this is also set as a rule in eslint:
image

It ensures us we won't import parent index into child within same location.

File: libs/shared/src/types/app-routes.ts (hunk 1 above)
Comment (Reden4005) [reply libs/shared/src/types/app-routes.ts]: Done.

File: libs/shared/src/components/localized-link/index.tsx (hunk 2 above)
Comment (Reden4005) [reply libs/shared/src/components/localized-link/index.tsx:6]: Indeed.

File: libs/shared/src/utils/create-path.ts (hunk 3 above)
Comment (Reden4005) [reply libs/shared/src/utils/create-path.ts:7]: Done.

**Pull request merged**